### PR TITLE
#121 add ses support

### PR DESCRIPTION
**Pull request by trestianp**

### CODE REVIEW

File: src/main/java/com/code4ro/legalconsultation/service/impl/AmazonEmailService.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AmazonEmailService {
+    private static final Logger LOG = LoggerFactory.getLogger(AmazonEmailService.class);
+
+    private static final String EMAIL_CHARSET = "UTF-8";
+    private static final String HTML_EMAIL_NOT_SUPPORTED_MESSAGE = "Your email client can't open this email. Contact us at contact@code4.ro";
```
Comment (trestianp) [inline src/main/java/com/code4ro/legalconsultation/service/impl/AmazonEmailService.java:16]: This is something I came up with. Let me know if you want to change it or leave it like this for now.

File: src/main/resources/application-production.yml [hunk 2]
```diff
@@ -20,25 +20,13 @@ spring:
...
-          auth: false # set the access user/password and change this to true
-          starttls:
-            enable: true
   cache:
     cache-names: users
 app:
   jwtSecret: ORy9HWgKoZPAs50ECKw0
   jwtExpirationInMs: 1728000000   # two days expiration time, can be extended
   signupurl: 'http://localhost:4200/authentication/sign-up'
+  emailSender: noreply@legalconsulting.ro # here should be the verified email in SES https://docs.aws.amazon.com/ses/latest/DeveloperGuide/verify-email-addresses-procedure.html
```
Comment (trestianp) [inline src/main/resources/application-production.yml:29]: This is something I came up with. Is there a real email we can setup ? 
In order for this to work, we need someone with AWS permissions to do the sender email setup.

File: src/main/java/com/code4ro/legalconsultation/service/impl/AmazonEmailService.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+import com.amazonaws.services.simpleemail.AmazonSimpleEmailService;
+import com.amazonaws.services.simpleemail.model.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AmazonEmailService {
```
Comment (mihairinzis) [inline src/main/java/com/code4ro/legalconsultation/service/impl/AmazonEmailService.java:12]: do we actually need this custom implementation? doesn't ses smtp work fine if we simply configure it in the properties file? i would prefer to have the email sending logic as generic as possible..

Comment (mihairinzis) [reply src/main/java/com/code4ro/legalconsultation/service/impl/AmazonEmailService.java:12]: it looks like ses supports smtp just fine: https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-smtp.html

Comment (trestianp) [reply src/main/java/com/code4ro/legalconsultation/service/impl/AmazonEmailService.java:12]: Yes. There is support for smtp.
My understanding was that we want to replace smtp.
If we want to use AWS SES servers with smtp, then the only change required is the servers in app properties and 'sender' . I will open a PR with that change.

Comment (mihairinzis) [reply src/main/java/com/code4ro/legalconsultation/service/impl/AmazonEmailService.java:12]: awesome @trestianp! thank you.

**Comment by mihairinzis** [comment]
we'll go smtp for now

**Pull request closed without merging**